src/charm.py

```diff
@@ -113,6 +116,9 @@ def __init__(self, *args: Any):
         if self._nginx_container.can_connect():
             self._set_alerts()
 
+        # do this regardless of what event we are processing
+        self._reconcile()
```

Comment:
while I agree that this is a good way to go, it makes me wonder why we didn't do the same in prometheus and loki non-HA and instead you chose to explicitly observe four different events?
Also there's more unconditional logic I see just above this line, that should be moved into _reconcile, but that feels like a refactoring that's easily out of scope for this PR. We should ask @lucabello or mike if they're fine with this change

Reply:
The reason I chose not to follow this pattern in non-ha charms was that it seemed that it'd need a whole lot of refactoring for the rest of the charm to follow the reconcile, so might as well just stick to what's already there

Reply:
for this charm, I refactored it to follow this reconcile pattern

Reply:
Yeah I think it's a good idea not to mix the two in charms that are too old-school